Sources/Features/Client/Model/AttachmentsStorage.swift

```diff
@@ -0,0 +1,131 @@
+import Foundation
+
+/// Type-alias of storing file attachments on disk (as a bookmark)
+/// Expected format: Filename, File URL bookmark
+public typealias Bookmarks = [String: NSData]
+
+final class AttachmentsStorage {
```

Comment:
It won't be easy to unify these since the function parameters are a bit different. Even the Configs are different (`AttachmentsStorage.swift` needs a different plist path than `AttributesStorage.swift`). Even so, I think it can be done, but it will involve significant refactoring. It's probably a good idea for us to do this, especially if we want to add other kinds of `Storage` classes for other kinds of report metadata.

Reply:
I think it looks much better now. Thank you for this good suggestion! Please see the updated version!